### DISCUSSION TITLE
Deploy application got error "Cannot find module '@/..." when run `next build` with filename's format is `kebab-case`

### DISCUSSION THREAD

**Discussion by nqhd3v** [vercel/next.js#46299] [Help]
### Verify canary release

- [ ] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:08:47 PST 2022; root:xnu-8792.61.2~4/RELEASE_X86_64
    Binaries:
      Node: 16.15.1
      npm: 8.11.0
      Yarn: 1.22.18
      pnpm: N/A
    Relevant packages:
      next: 13.1.6
      eslint-config-next: 13.1.6
      react: 18.2.0
      react-dom: 18.2.0

warn  - Latest canary version not detected, detected: "13.1.6", newest: "13.1.7-canary.27".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue

### Link to the code that reproduces this issue

https://github.com/nqhd3v/beta.nqhuy

### To Reproduce

First of all, I think this is not a coding bug, because, in the local environment, I have been running `yarn build` or `yarn start` OK. And for system admin only (to track logs on their system), so this problem can't be reproduced (if possible, clone this source, and re-deploy on your vercel)

### Describe the Bug

when I deploy it with Vercel, it got some error like this:

[Check this deploy logs here](https://vercel.com/nqhd3v/beta-nqhuy/HWWeBzXdbkt2QC86gFeoC3h8bGda).  
<img width="993" alt="image" src="https://user-images.githubusercontent.com/106160187/220804081-3ad79f64-a517-4fc9-8052-b260423d500d.png">

[Check this deploy logs here](https://vercel.com/nqhd3v/beta-nqhuy/HxML9LZqUGSEE6GZHrBXnrieB7q4).  
<img width="999" alt="image" src="https://user-images.githubusercontent.com/106160187/220803929-f7324bf1-bb05-4b40-b56f-ab10d4876b3e.png">

When I got this issue, I think, maybe because of the naming file (uppercase, or downcase), this issue usually happened on macOS. But, it is normal, next, I think, maybe because of  the `tsconfig` file `@/..` instead of `../somethings/..`, but it is OK too :) I compared it with your docs, and it supports next@9.3+ (my ver is 13).

And now, the last solution, I remove `dash` for my filename, and it worked! I don't know why, because, in my source, I used `-` in my filename somewhere.

[For VERCEL system | Check last deploy's logs here](https://vercel.com/nqhd3v/beta-nqhuy/E9VvTWNry4Gs7BLbinqLcJ85XZqE) or [last commit here](https://github.com/nqhd3v/beta.nqhuy/commit/c4e9fa33494d4b42da12bc1d0fbeaa671ae84be1)

### Expected Behavior

Absolutely, my expectation is it ready after I run deploy!

**Comment by balazsorban44** [comment]
Hi, I tried following the reproduction steps, and I was able to deploy without a problem. See: https://balazsorban-46279.vercel.app

Could you try again to make sure it wasn't a temporary fluke?

**Reply by nqhd3v** [reply]
are you sure you are implementing the correct commit that I mentioned? Try again one more time, with branch `fluke-1` or `fluke-2` in my source. I retry, and still errors in my logs.  
https://github.com/nqhd3v/beta.nqhuy

<img width="1219" alt="image" src="https://user-images.githubusercontent.com/106160187/220964675-a68909d7-3655-49b3-b3f7-0bea01a8e320.png">
